pkg/cloud/azure/services/certificates/certificates.go

```diff
@@ -40,24 +42,25 @@ import (
 	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
 	"k8s.io/kubernetes/cmd/kubeadm/app/util/pubkeypin"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/services"
 )
 
 // Get should implement returning certs and kubeconfigs.
-func (s *Service) Get(ctx context.Context) (interface{}, error) {
+func (s *Service) Get(ctx context.Context, spec interface{}) (interface{}, error) {
```

Comment:
Do we really want to pass a generic `interface{}` type as part of the function signature?

Reply:
Yep! changing in next PR